docs: developer incentives, register cosmos coins (#791)
evmos/evmos: docs/developers/guides/incentives_registration.md

<!--
order: 5
-->

# Smart Contract Incentive Registration

Learn how to register smart contract incentives through [Evmos Governance](../../users/governance/overview.md). {synopsis}

> Smart contracts can register incentives through governance. After incentives are registered for a smart contract, users that interact with the contract (eg. through submission of transactions) will have rewards distributed to them. These rewards are essentially cash-back for the gas spent on the initial transactions.

Reviewer: ```suggestion
An incentive for a smart contract can be registered through governance to encourage more interaction with that contract. Users who interact with a registered contract (eg. through submission of transactions) will have rewards distributed to them. These rewards are essentially cash-back for the gas spent on the initial transactions.
```

The incentives workflow is summarized in the following image:

![workflow](./../../img/incentives_workflow.png)

To register smart contract incentives, consider the following stages:

1. [Drafting the Contract Incentives Proposal](#drafting-the-contract-incentives-proposal)
2. [Submitting the Contract Incentives Proposal](#submitting-the-contract-incentives-proposal)
3. [The On-Chain Contract Incentives Proposal](#the-on-chain-contract-incentives-proposal)

This document also discusses [incentives analysis through telemetry](#incentives-analysis-through-telemetry).

> 

Reviewer: Add like to spec for more info

### Emitting & Collecting Metrics

If telemetry is enabled via configuration, a single global metrics collector is exposed via the [`go-metrics`](https://github.com/armon/go-metrics) library. This allows for emitting and collecting metrics through a simple [API](https://github.com/cosmos/cosmos-sdk/blob/v0.46.0-rc1/telemetry/wrapper.go). For example:

```go
func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
  defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)

  // ...
}
```

Developers may use the [`telemetry`](https://docs.evmos.org/protocol/telemetry.html) package directly, which provides wrappers around metric APIs that include adding useful labels, or they must use the `go-metrics` library directly. It is preferable to add as much context and adequate dimensionality to metrics as possible, so the `telemetry` package is advised.

> Regardless of the package or method used, the Cosmos SDK supports `guages`, `summaries`, and `counters` as metric types.

Reviewer 2: this is not relevant for a developer since they won't modify the golang code, they just need to know how to use the telemetry
